scripts/test_adv_IM.py

```diff
@@ -14,23 +24,81 @@
     )
 
+    parser.add_argument(
+        "--components",
+        default=[const.Components.c000.str_value, const.Components.c090.str_value],
+        nargs="+",
+        choices=list(const.Components.iterate_str_values()),
+        help="list of component that ran",
+    )
+
     args = parser.parse_args()
 
+    return args
+
+
+def main(sim_dir, adv_im_model, components, exit=True):
+
     failed = []
-    for model in args.adv_im_model:
-        csv_path = os.path.join(args.sim_dir, "IM_calc/{}.csv".format(model))
+
+    for model in adv_im_model:
+        # check if csv exist
+        csv_path = os.path.join(sim_dir, "IM_calc/{}.csv".format(model))
+        im_dir = get_im_calc_dir(sim_dir)
         # check file exist first to prevent crash
         try:
             df = pd.read_csv(csv_path)
         except FileNotFoundError:
             failed.append(model)
             print("csv for {} does not exist".format(model))
         else:
+            # check for null/nan
             if df.isnull().values.any():
                 failed.append(model)
+
+            # if csv exist:
+            # check station count, if match, skip rest of checks
+            csv_station_count = len(df.station.unique())
+            # glob for station folders
+            list_folders = [x for x in os.listdir(im_dir) if os.path.isdir(x)]
+            if csv_station_count == len(list_folders):
+                # station folder count matches csv.station.count
+                continue
+            else:
+                for station_dir in list_folders:
+                    # check if folder is a station run folder
+                    station_model_dir = os.path.join(station_dir, model)
+                    station_name = os.path.basename(station_dir)
+                    if not os.path.isdir(station_model_dir) or (
+                        station_name in df.station.unique()
+                    ):
+                        # does not match naming for a station run, possibly created by something else, skipping
+                        continue
+                    else:
+                        result = True
+                        for comp in components:
+                            component_outdir = os.path.join(station_model_dir, comp)
+                            result = (
+                                check_status(component_outdir, check_fail=True)
+                                and result
+                            )
+                        # check of "Failed"
+                        # no "Failed" in all Analysis*.txt == crashed
+                        # "Failed" in all Analysis*.txt == failed to converge (normal), continue
+                        if result != True:
+                            # logs shows not all analysis failed to converge, but station failed to create csv
+                            print(f"{sim_dir}: {station_name} failed to run on {model}")
+                            failed.append(model)
```

Comment:
I suggest using a break in here, instead of continuing to check after we know it's failed e.g.
`if not check_status: print(); append(); break`

Reply:
What I'd recommend is breaking this into separate functions. It seems like this function has too much going on. What is the logical flow of the program and go from there. 

At least a comment in the docstring saying what testing steps are being done here.

Reply:
the reason of not using a break is that it shows you all the stations that had failed the analysis. So you can look at them all at once. otherwise it can potentially waste a lot of CH if you only realized that there is another station crashed after you run the whole event.

Reply:
Using break will not stop after the first station. It will stop after the first broken component for each station.
The bit is saves on is calling check_status after something has already failed.

I agree with breaking this function up though

Reply:
the line you commented on is looping at station level, instead of comp within a station; hence the confusion. 
but i do agree on breaking when one component fails.